docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -39,12 +39,30 @@ impl Storage {
 If you have defined a `Storage` struct following this naming scheme, then it will be made available to you through the reserved `storage` keyword within your contract functions.
 
 :::warning Using slot `0` is not supported!
-No storage values should be initialized at slot `0`. This is a known issue that will be fixed in the future.
+No storage values should be initialized at slot `0` - storage slots begin at `1`. This is a known issue that will be fixed in the future.
 :::
 
+## Storage Slots
+
+Internally, Aztec keeps contract state in a storage slot key value store of type `Map<Field, Field>`, stored as a merkle tree(?). For now, storage slot positions for each variable must be explicitly assigned inside the `Storage impl`. Except for `Map` types, storage is in contiguous blocks, so you must calculate how many slots each public variable requires, and increment the next variable's slot by this amount.
```

Comment:
Might be worth saying all public state is stored within one merkle tree, where contract's act as a domain seperator.

Do we really need to increment the next slot by this amount? The final slot e.g. (1, address) will be hashed giving a random value. Longer slots are then added to to there. 

Assuming the hash is sufficiently random slot (2, address) will give another completely random value. Therefore we can relatively safely store 3 blocks of data @ { hash(2, address), h(2, address) + 1, h(2, address) + 2) as it will be no where near hash(3, address)

Im not fully confident on this